File: code_eval/java/easy/Challenge113.java
package easy;

import java.io.BufferedReader;
import java.io.File;
import java.io.FileReader;

/**
 * Challenge 113 : Multiply Lists
 * Created by cschalk on 4/11/16.
 */
public class Challenge113 {

    public static void main (String[] args) throws Exception {
        File file = new File(args[0]);
        BufferedReader buffer = new BufferedReader(new FileReader(file));
        String line;
        while ((line = buffer.readLine()) != null) {
            line = line.trim();
            runChallenge(line);
        }
    }

    private static void runChallenge(String input) {
        String sets[] = input.split("\\|");
        String set1[] = sets[0].trim().split(" ");
        String set2[] = sets[1].trim().split(" ");
        for (int i = 0 ; i < set1.length ; ++i) {
            int ans = Integer.parseInt(set1[i]) * Integer.parseInt(set2[i]);
            System.out.print(ans + " ");
        }
        System.out.println();
    }
}

File: code_eval/cpp/easy/challenge_113.cpp
// Created by cschalk on 4/6/16.
// Challenge 113 : Multiply Lists
#include <iostream>
#include <fstream>
#include <sstream>

using namespace std;

void multiplySets(const string &set1, const string &set2) {
    stringstream sset1(set1);
    stringstream sset2(set2);
    int one, two;
    while (sset1 >> one) {
        sset2 >> two;
        cout << one * two << " ";
    }
    cout << endl;
}

int main(int argc, char ** argv) {
    ifstream input {argv[1]};
    string line;
    while (getline(input, line)) {
        int token = line.find_first_of('|');
        string set1 = line.substr(0, token++);
        string set2 = line.substr(token, line.length() - token);
        multiplySets(set1, set2);
    }
}

